pkg/configuration/config_manager/config_handler.go

```diff
@@ -169,37 +171,77 @@ type shellCommandHandler struct {
 	downwardAPIMountPoint []string
 	downwardAPIHandler    map[string]ConfigHandler
 
-	backupPath string
-	configMeta *ConfigSpecInfo
-	filter     regexFilter
+	backupPath    string
+	configMeta    *ConfigSpecInfo
+	filter        regexFilter
+	isBatchReload bool
 }
 
 func (s *shellCommandHandler) OnlineUpdate(ctx context.Context, name string, updatedParams map[string]string) error {
 	logger.V(1).Info(fmt.Sprintf("online update[%v]", updatedParams))
 	logger.Info(fmt.Sprintf("updated parameters: %v", updatedParams))
 	args := make([]string, len(s.arg))
-	return s.execHandler(ctx, updatedParams, args)
+	copy(args, s.arg)
+	_, err := s.execHandler(ctx, updatedParams, args)
+	return err
 }
 
-func (s *shellCommandHandler) execHandler(ctx context.Context, updatedParams map[string]string, args []string) error {
-	commonHandle := func(args []string) error {
+func (s *shellCommandHandler) execHandler(ctx context.Context, updatedParams map[string]string, args []string) ([]string, error) {
+	if s.isBatchReload {
 		command := exec.CommandContext(ctx, s.command, args...)
+		stdin, err := command.StdinPipe()
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot create a pipe connecting to the STDIN of the command")
+		}
+		var stdinBuilder strings.Builder
+		go func() {
+			defer stdin.Close()
+			for key, value := range updatedParams {
+				line := fmt.Sprintf("%s\t%s\n", key, value)
```

Comment:
The format of this batch is too solidified.

The use of the scene is limited, It is recommended to use template to define the batch format.

e.g:

```
{{- range $pKey, $pValue := $.arg0 }}
{{ printf "%s\t%s" $pKey $pValue }}
{{- end }}
```

Reply:
hi @sophon-zt 
 Can you enumerate a few more examples of templates?

Reply:
Refer to this implementation: https://github.com/apecloud/kubeblocks/blob/main/pkg/configuration/config_manager/reload_util.go#L67

reference code：
```
engine := gotemplate.NewTplEngine(gotemplate.ConstructFunctionArgList(updatedParams, s.args...), nil, "render-batch-action-parameters", nil, nil, gotemplate.WithCustomizedWithType(gotemplate.KBDSL))
## batchParametersTemplate is shellTrigger.batchParametersTemplate
stdinString, err := engine.Render(batchParametersTemplate)
if err != nil {
    ## process error
}
# write stdin
io.WriteString(stdin, stdinString)
```

example, update parameters: field1=value1,field2=value2,field3=value3

batchParametersTemplate:
```
{{- range $pKey, $pValue := $.arg0 }} 
{{ printf "%s:%s" $pKey $pValue }}
{{- end }}
```

render result:
```
field1:value1
field2:value2
feidl3:value3
```

Reply:
Fixed in da1971c221782b17a9fa4925ad56d033e04cc925

Reply:
Finally I choose to define a tempale like this:
```
{{- range $pKey, $pValue := $ }}
{{ printf "%s=%s" $pKey $pValue }}
{{- end }}
```
I use `$` instead of `$.arg0` , and pass the updated params as a map referred as `$` .  Because I felt that `$.arg0` was too obscure for users, who might not know where `arg0` came from